software-mansion/react-native-reanimated
# Memoize `_getViewInfo` result

## Summary

This PR adds memoization of `_getViewInfo` result in `createAnimatedComponent` in order to avoid calling it on each re-render. On Fabric, the expected behavior is that `getShadowNodeWrapperFromRef` will be called only once after the component is mounted.

Additionally, this PR reverts changes introduced in #4986 because now `ShadowNode` is read only once and never changes so the condition is always false.

## Test plan

1. Add `console.log` in `getShadowNodeWrapperFromRef`
2. Open `WidthExample.tsx`
3. Click "Increase height"/"Increase padding" buttons to force renders
4. There should be no logs in the output
